controllers/destinationProxy.js

```diff
@@ -0,0 +1,110 @@
+const networkHandlerFactory = require('../adapters/networkHandlerFactory');
+const { getPayloadData } = require("../adapters/network");
+const { TRANSFORMER_METRIC } = require("../v0/util/constant");
+const {
+  generateErrorObject,
+} = require("../v0/util");
+const stats = require("../util/stats");
+const match = require("match-json");
+const jsonDiff = require("json-diff");
+const logger = require("../logger");
+
+class DestProxyController {
+  /**
+   * Handler for testing the destination proxy
+   * @param {*} destination Destination name
+   * @param {*} ctx 
+   * @returns 
+   */
+  static handleProxyTestRequest = async (destination, ctx) => {
+    const { deliveryPayload: routerDeliveryPayload, destinationRequestPayload: routerDestReqPayload } = ctx.request.body;
+    let response;
+    try {
+      const destNetworkHandler = networkHandlerFactory.getNetworkHandler(
+        destination
+      );
+      
+      const proxyDestReqPayload = destNetworkHandler.prepareProxy(routerDeliveryPayload);
+      response = {
+        destinationRequestPayload: proxyDestReqPayload,
+      };
+
+      // Special handling required as Go and JavaScript encodes
+      // URL parameters differently
+      const { payloadFormat } = getPayloadData(routerDeliveryPayload.body);
+      if (payloadFormat === "FORM") {
+        // This is to make sure we encode `~` in the data coming from the router
+        // The result coming from the router is already a query parameter string
+        const routerBodyVal = new URLSearchParams(routerDestReqPayload.body);
+        routerDestReqPayload.body = routerBodyVal;
+  
+        const proxyBodyVal = new URLSearchParams();
+        proxyDestReqPayload.body.forEach(function(value, key) {
+          const encodeAsterisk = (x) => x.replace(/\*/g, "%2A");
+          // Router encodes `*` as well
+          proxyBodyVal.append(encodeAsterisk(key), encodeAsterisk(value));
+        });
+        proxyDestReqPayload.body = proxyBodyVal;
```

Comment:
If you see the return of `prepareProxyRequest`, there is no `body` property at all. Why are we making use of an unknown property in comparison ?

Reply:
Fixed it.